src/Guzzle/StatusDateOverrideMiddleware.php

```diff
@@ -0,0 +1,110 @@
+                    case 'application/vnd.elife.article-history+json; version=1':
+                        $data['versions'] = $this->updateItems($data['versions']);
+                        break;
+
+                    case 'application/vnd.elife.article-list+json; version=1':
+                        $data['items'] = $this->updateItems($data['items']);
+                        break;
+
+                    case 'application/vnd.elife.article-related+json; version=1':
+                        $data = $this->updateItems($data);
+                        break;
+
+                    case 'application/vnd.elife.article-poa+json; version=1':
+                    case 'application/vnd.elife.article-poa+json; version=2':
+                    case 'application/vnd.elife.article-vor+json; version=1':
+                    case 'application/vnd.elife.article-vor+json; version=2':
+                    case 'application/vnd.elife.article-vor+json; version=3':
+                        $data = $this->updateItem($data);
+                        break;
+
+                    case 'application/vnd.elife.collection+json; version=1':
+                        $data['relatedContent'] = $this->updateItems($data['relatedContent'] ?? []);
+                        break;
+
+                    case 'application/vnd.elife.cover-list+json; version=1':
+                    case 'application/vnd.elife.highlight-list+json; version=3':
+                        $data['items'] = array_map(function (array $cover) {
+                            $cover['item'] = $this->updateItem($cover['item']);
+
+                            return $cover;
+                        }, $data['items']);
+                        break;
+
+                    case 'application/vnd.elife.podcast-episode+json; version=1':
+                        $data['chapters'] = array_map(function (array $chapter) {
+                            $chapter['content'] = $this->updateItems($chapter['content'] ?? []);
+
+                            return $chapter;
+                        }, $data['chapters']);
+                        break;
+
+                    case 'application/vnd.elife.press-package+json; version=3':
+                        $data['relatedContent'] = $this->updateItems($data['relatedContent'] ?? []);
+                        break;
+
+                    case 'application/vnd.elife.search+json; version=1':
+                        $data['items'] = $this->updateItems($data['items']);
+                        break;
+                }
+
+                return $response->withBody(stream_for(json_encode($data)));
+            });
+        };
+    }
+
+    private function updateItems(array $items) : array
+    {
+        return array_map([$this, 'updateItem'], $items);
+    }
+
+    private function updateItem(array $item) : array
+    {
+        if (isset($item['statusDate']) && isset($this->rdsArticles[$item['id']]['date']) && $this->authorizationChecker->isGranted('FEATURE_RDS')) {
```

Comment:
```suggestion
        if (isset($item['statusDate']) && isset($this->rdsArticles[$item['id']]) && $this->authorizationChecker->isGranted('FEATURE_RDS')) {
```
as `date` should be mandatory by now for each of the RDS articles.

Reply:
If we can assume that any entry for `rds_articles` in the configuration contains a date then this makes sense. Will make this refactor.